### PR TITLE
Subscritpions manual test plan

### LINKED ISSUE #46: [Eventing - Control Plane] Subscription Lifecycle Test Plan
### Description
---

This issue describes the creation of a "Manual Testing Plan" for a portion of the Eventing Spec.

Below is a link to the Spec Sub-heading which this issue is referencing, alongside a template upon which you should base your PR.

**Link to Spec**: https://github.com/knative/specs/blob/main/specs/eventing/control-plane.md#subscription-lifecycle

**Template**: https://hackmd.io/D1S9NPaRTG2dcyFtdURSww?view

If you have any questions, please reach out in the #conformance channel on Slack

---

### Acceptance Criteria
---
- [ ] Contributing to the right place

GIVEN I am accepting the PR that closes this issue 
WHEN I inspect the .md file 
THEN I see it is being committed to the /test/test-plan/ folder in the specs repo

- [ ]   Conformance to Template

GIVEN I am reviewing the submitted .md file 
WHEN I compare it to the template linked above 
THEN I see that the submitted .md conforms to the template 

- [ ]  Coverage of Spec

GIVEN I am reviewing the submitted .md file
WHEN I compare the .md file to the Subscription Lifecycle subheading in the spec
THEN all of the  "MUST" "MUST NOT" and "REQUIRED" sentences under the Subscription Lifecycle sub-heading are covered within the submitted .md file 

- [ ] Coverage of Tests

GIVEN I am running each of the tests in the submitted .md file 
AND each test has a expected output
WHEN I am finished running the tests 
THEN all of my tests pass (when compared to the expected output)
AND at least 2 reviewers feel it is properly covering all portions of the spec outlined at the top .md file

### PR DESCRIPTION
**Pull request by gabo1208**
Fixes #46

### CODE REVIEW

File: test-plan/eventing/subscription-lifecycle-conformance.md [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+```
+ kubectl get subscription conformance-subscription -ojsonpath="{.status.conditions[?(@.type == \"Ready\")].status}"
+```
+
+Lets now check the reason why the Subscription is not ready yet:
+
+```
+ kubectl get subscription conformance-subscription -ojsonpath="{.status.conditions[?(@.type == \"Ready\")].reason}"
+```
```
Comment (odacremolbap) [inline test-plan/eventing/subscription-lifecycle-conformance.md:120]: Conformance states that at least one condition MUST indicate the reason for the error.
I think that checking the at least one of the other conditions is set to false would make this conformant. Would there by some kubectl + JSON able to test that?

Comment (odacremolbap) [reply test-plan/eventing/subscription-lifecycle-conformance.md:120]: Also, at this stage it should be checked that `status.physicalSubscription` is empty

File: test-plan/eventing/subscription-lifecycle-conformance.md [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+## [Pre] Creating the DeadLetterSink and the Subscriber
+
+```
+kubectl apply -f control-plane/subscription-lifecycle/services.yaml
+```
+
+## [Test] Subscription status update
+
+The `status.physicalSubscription` should have been updated with the
```
Comment (odacremolbap) [inline test-plan/eventing/subscription-lifecycle-conformance.md:146]: Unfinished sentence

File: test-plan/eventing/subscription-lifecycle-conformance.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+```
+{
+  "test": "control-plane/subscription-lifecycle/subscription-readiness-2"
+  "output": {
+	  "expectedType": "Ready",
+	  "expectedStatus": "True"
+  }
+}
+```
```
Comment (odacremolbap) [inline test-plan/eventing/subscription-lifecycle-conformance.md:190]: I am not sure what is being tested here. The subscription is now Ready and I guess that this test is about that but the name of the section ([Test] Subscription Readiness 2) and the first paragraph does not clarify it.

Can those be rephrased?

File: test-plan/eventing/subscription-lifecycle-conformance.md [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+
+# Clean up & Congratulations
+
+Make sure that you clean up all resources created in these tests by running (ignore the errors :p): 
+
+```
+kubectl delete -f control-plane/subscription-lifecycle/
```
Comment (odacremolbap) [inline test-plan/eventing/subscription-lifecycle-conformance.md:253]: I think there are only 4 objects to delete so we could put up a one liner command to do so. What about adding a label to all objects at manifests, for example `app=conformance` and then cleaning by issuing a
```
kubectl delete all -l app=conformance
```
WDYT?

**Comment by evankanderson** [comment]
/close

This was an exercise in 1.0 preparation, but we don't need to commit these files at this point.

**Pull request closed without merging**